### PR TITLE
Address `JS` to `DotNet` byte[] Interop API Review Feedback

### PR DESCRIPTION
**Pull request by TanayParikh**
Fixes: https://github.com/dotnet/aspnetcore/issues/34327

### CODE REVIEW

File: src/Components/Server/src/Circuits/RemoteJSDataStream.cs [hunk 1]
```diff
@@ -40,22 +40,20 @@ public static async Task<bool> ReceiveData(RemoteJSRuntime runtime, long streamI
             RemoteJSRuntime runtime,
             IJSStreamReference jsStreamReference,
             long totalLength,
-            long maximumIncomingBytes,
+            long signalRMaximumIncomingBytes,
```
Comment (TanayParikh) [inline src/Components/Server/src/Circuits/RemoteJSDataStream.cs:43]: This change wasn't part of the API review, but I just felt this updated name could avoid future confusion (ex. clarifying it's not the max incoming bytes for the whole stream somehow).

Comment (SteveSandersonMS) [reply src/Components/Server/src/Circuits/RemoteJSDataStream.cs:43]: Great clarification. I probably *would* have thought it referred to the stream length before :)

**Review by SteveSandersonMS — APPROVED** [review]
I don't think I was in the review that must have covered it, so was the conclusion that we shouldn't expose configurability over the `Pipe` options because the defaults are always good? If so, I'm certainly happy about that. If the reasoning or goal is something different from that, please let me know because I'd love to fully understand.

Approving assuming this is the case :)

**Comment by pranavkm** [comment]
> I don't think I was in the review that must have covered it, so was the conclusion that we shouldn't expose configurability over the Pipe options because the defaults are always good

The defaults are reasonable, any configuration here isn't directly visible because the abstraction the user works with is a `Stream`, and Kestrel felt that they made a mistake exposing the pipeoptions to users in the few places they did since it's very easy to misconfigure it.

**Review by pranavkm — APPROVED** [review]

**Pull request merged**